target/linux/pistachio/base-files/lib/functions/mac.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+#
+# Copyright 2016 by Imagination Technologies Limited and/or its affiliated group companies.
+#
+# All rights reserved.  No part of this software, either
+# material or conceptual may be copied or distributed,
+# transmitted, transcribed, stored in a retrieval system
+# or translated into any human or computer language in any
+# form by any means, electronic, mechanical, manual or
+# other-wise, or disclosed to the third parties without the
+# express written permission of Imagination Technologies
+# Limited, Home Park Estate, Kings Langley, Hertfordshire,
+# WD4 8LZ, U.K.
+#
+# common shell lib to obtain wifi mac address
+
+
+get_primary_macaddr()
+{
+	local wifi_mac_addr=0x$(hexdump -n 6 -v -e '/1 "%02X"' /proc/device-tree/uccp@18480000/mac-address0)
+	if [ $? -eq 0 ]
+	then
+		echo $wifi_mac_addr
+	fi
+}
+
+get_lowpan_mac()
+{
+	local PRIMARY_MAC_ADDR=$(get_primary_macaddr)
+	if [ ! -z $PRIMARY_MAC_ADDR ]
+	then
+		# 6LOWPAN_MAC = PRIMARY_MAC_ADDR + 5
+		local LOWPAN_OTP_OFFSET=0x5
+		local LOWPAN_MAC=$(printf "%012X" $(($PRIMARY_MAC_ADDR + $LOWPAN_OTP_OFFSET)))
+		LOWPAN_MAC=$(macaddr_setbit_la $(macaddr_canonicalize $LOWPAN_MAC))
+		echo $LOWPAN_MAC
+	fi
+}
+
+get_bluetooth_mac()
+{
+	local PRIMARY_MAC_ADDR=$(get_primary_macaddr)
```

Comment:
@francois-berder-imgtec  I was talking about the way they were implemented, the current implementation for both returns the MAC address